### PR TITLE
Liquidator analysis

### PR DESCRIPTION
**Pull request by james001-c**
kindly review

### CODE REVIEW

File: notebooks/Liquidators_analysis.ipynb [hunk 1]
```diff
@@ -0,0 +1,1351 @@
...
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Group by 'from_address' and count the number of unique loans liquidated by each liquidator\n",
+    "liquidators_loan_counts = ep_data.groupby('from_address')['transaction_hash'].nunique().reset_index()\n",
```
Comment (lukaspetrasek) [inline notebooks/Liquidators_analysis.ipynb:632]: No, this is the number of transactions per smart contract.

File: notebooks/Liquidators_analysis.ipynb [hunk 2]
```diff
@@ -0,0 +1,1351 @@
...
+      "text/plain": [
+       "<Figure size 1000x600 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# Group by 'User' and 'Protocol' to get the number of liquidators per protocol\n",
```
Comment (lukaspetrasek) [inline notebooks/Liquidators_analysis.ipynb:213]: Hint: Check the values of the `key_name` column in the data. You'll find some liquidation events there.

File: notebooks/Liquidators_analysis.ipynb [hunk 3]
```diff
@@ -0,0 +1,1351 @@
...
...EiRNrXbd169afdQQEBNS69pdAaGgoOnXqhKNHj2Lw4MFo3LgxQkJCUFZWhlevXqFnz54IDg5mom1jY6PgvTt+/HjejxionM8YGxsz0QaAr4EpNaHuttZbt26FtbU1EhMT+XOsA1P5+fkoLy9n9vP/CT8/P8TExODly5eIj4+HlZUVYmJi+PEff/wRvr6+TLQbNmyImTNnAqh8kBsYGGDGjBn8+Pz58+Hl5cVEW55Bgwaha9eueP/+PXR0dFBUVIRbt27B29sbo0aNYqa7dOlSBAYGolevXvxnXsajR49gb2/PTBsAOnXqhNDQUDx8+BA6Ojq4fPkysrOz0bRpUxw5coSZrqmpKc6fP69wbtiwYbCxscGNGzdUFpiS58iRI+jXrx+kUimkUikTXflDPhAoHwxkuWDjOA6XLl3CokWL0KBBAwgEAnh6eiIxMZFfVLFCnQt0GRKJROmi6erVq5BIJMx05Sexs2bNgomJCf744w/cu3cPO3bsQN26dRXue/8WHjx4gOLiYrVoGxgY8EbnhoaGEIlEEAgEfFtt2b8NDQ3Vcn2qwMvLize7ll+snjt3DmZmZsz1w8PD4evri9zcXEilUmRmZiI1NRXOzs7466+/mGp//PgRc+fOhaWlJX+PtbS0xNy5c5nMt8RiMd/AQB2BIXWbn8vw9/fH6tWrq51PTU1lFvz/8OEDRo4cCYlEAo7joKWlBZFIBI7jIJFIMGrUKLx//56J9pdC7969MWjQIADg57AvX75EmzZtVNJAaPny5bCxseG/a1ZWVli5ciVzXRnp6enw9fXl7/e+vr7Mm7moK/j7X+batWtwcnICx3Fwd3fHvXv30KVLF2hqakJTUxMmJiY4c+YME+0uXbpg8eLFNY7/+uuvaNOmDRNt4GtgSi2kpqZCU1MTDRs2hI6ODlJSUmBgYIDo6GhERUVBJBIhIyOD+XXcvXsXbdq0QXBwMB48eMA8MCUQCPDw4UMAgL29PR4/fsxMSxl6enq4fv06gMoHvKamJs6dO8ePX7t2Dfr6+ky0pVIpv4sm6+wgH7C4ceMG8850AFBaWgo/Pz8YGBhAQ0MD1tbWEAqF8Pf3Z7bTlZCQAG1tbcTExKBv377Q0tJCXFwcP84yOCPDyMgI+fn5ACo/B1euXAEAHDhwgGlAUFdXV2nXouHDh8PKygpHjhxh9rv/06KhtLQUy5c
```
Comment (lukaspetrasek) [inline notebooks/Liquidators_analysis.ipynb:673]: I see a lot of cells likely used for debugging below this one. Pls remove them, or add some comments explaining why do we need to keep these cells.

Comment (lukaspetrasek) [inline notebooks/Liquidators_analysis.ipynb:676]: The rest of the analysis is completely wrong. Do not hesitate to ask if you don't know something 😉

Comment (lukaspetrasek) [inline notebooks/Liquidators_analysis.ipynb:672]: Compute the amount of liquidated debt like this:
1) Identify only liquidation events. Use Starkscan to explore their structure.
2) Identify the amounts of debt tokens liquidated. Optionally, analyze the collateral tokens liquidated for extra points 🥇
3) Compute aggregate states, like total volumes liquidated, total amounts per token, average liquidated amounts, etc.
4) [Optional] Fetch prices for these tokens (in USD). E.g. from Coingecko.
5) [Optional] Multiply the token amount by the corresponding prices.
6) Repeat task 3) for the USD amounts.

**Review by lukaspetrasek — CHANGES_REQUESTED** [review]

**Comment by lukaspetrasek** [comment]
Hi @james001-c  ,

We'll be merging this PR so that it counts towards the OD Hackaton. You will receive a partial reward, but we'll be happy if you open a new PR and finish the task!

**Comment by james001-c** [comment]
Hello CarmineOptions/derisk-research

I stopped working because you dropped a message on the telegram account
that you were going to merge the work so I thought I lost already. Please
kindly re-assign me.

**Pull request merged**